cmd/oras/attach.go

```diff
@@ -19,24 +19,25 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"strings"
 
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/spf13/cobra"
 	"oras.land/oras-go/v2"
 	"oras.land/oras-go/v2/content"
 	"oras.land/oras-go/v2/content/file"
+	"oras.land/oras-go/v2/registry/remote"
 	oerrors "oras.land/oras/cmd/oras/internal/errors"
 	"oras.land/oras/cmd/oras/internal/option"
 )
 
 type attachOptions struct {
 	option.Common
-	option.Remote
 	option.Packer
 	option.ImageSpec
 	option.DistributionSpec
```

Comment:
I just noticed that the `DistributionSpec` should be stick to the `option.Remote`. This should also be applied to `oras cp` and `oras manifest push`.

Reply:
No it should not. If we do that, `DistributionSpec` will apply flag to all commands related to `option.Remote`, whichi is wrong considering the following cases:
1. Since `oras push` is not related to referrers maintanance, the execution is always the same no matter what value is provided to `--distribution-spec`. So we should NEVER add this flag to those commands.
2. If user specify `--distribution-spec v1.1-referrers-tag` to `oras discover`, should the command looking for referrers via tag scheme? We should design the expected behaviour and then add this flag to those commands.